nautobot_secrets_providers/providers/keeper.py

```diff
@@ -0,0 +1,188 @@
+"""Secrets Provider for Keeper."""
+import os
+# from pathlib import Path
+# import base64
+# import json
+
+try:
+    from keeper_secrets_manager_core import SecretsManager
+    from keeper_secrets_manager_core.core import KSMCache
+    from keeper_secrets_manager_core.exceptions import KeeperError, KeeperAccessDenied
+    from keeper_secrets_manager_core.storage import FileKeyValueStorage #, InMemoryKeyValueStorage
+    # from keeper_secrets_manager_core.utils import get_totp_code
+except (ImportError, ModuleNotFoundError):
+    keeper = None
+
+from django import forms
+from django.conf import settings
+# from django.core.exceptions import ValidationError
+
+from nautobot.apps.secrets import exceptions, SecretsProvider
+from nautobot.utilities.forms import BootstrapMixin
+
+from .choices import KeeperTypeChoices
+
+
+__all__ = ("KeeperSecretsProvider")
+
+
+try:
+    plugins_config = settings.PLUGINS_CONFIG["nautobot_secrets_providers"]
+    KEEPER_TOKEN = plugins_config["keeper"]["token"]
+except KeyError:
+    KEEPER_TOKEN = None
+
+class KeeperSecretsProvider(SecretsProvider):
+    """A secrets provider for Keeper Secrets Manager."""
+
+    slug = "keeper-secret-manager"
+    name = "Keeper Secret Manager"
+
+    class ParametersForm(BootstrapMixin, forms.Form):
+        """Parameters for Keeper Secrets Manager."""
+
+        name = forms.CharField(
+            label="Secret Name",
+            help_text="The secret's name",
+            max_length=30,
+            min_length=5,
+        )
+        uid = forms.CharField(
+            label="Secret UID",
+            help_text="The secret's uid",
+            max_length=25,
+            min_length=20,
+        )
+        token = forms.CharField(
+            label="Token",
+            widget=forms.PasswordInput,
+            help_text="The One Time Token",
+            max_length=40,
+            min_length=20,
+            initial=KEEPER_TOKEN,
+        )
```

Comment:
I'm hesitant to have this as a user-specified form field since its value would be user-readable and stored in the DB. Isn't the token a sensitive value?

Reply:
I agree with you wit traditional token
With Keeper Secrets Manager it's a One Time Token that will generate the config file config.json that will be used afterwards
I'm still unsure of the best way to process this, so I implement both way of providing connection, either with the config in JSON or BASE64 (not implemented yet as seen on #L156 or with the Token that will be valid once only